### ISSUE TITLE
jsx-handler-names not works well with stateless components

### ISSUE THREAD

**Issue by playpauseandstop** [yannickcr/eslint-plugin-react#346]
I enabled `react/jsx-handler-names` rule today for my codebase and everything went well, except one case with stateless component for modal. I put all handlers there in props and from docs it shouldn't consider warnings, but it does. 

Next code,

```js
import React, {PropTypes} from "react";
import {Button, Modal} from "react-bootstrap";

const ClearChangesModal = (props) => {
    return (
        <Modal onHide={props.onClose} show>
            <Modal.Body>
                <p>Are You Sure to discard all changes?</p>
                <p className="text-danger">This action is undone.</p>
            </Modal.Body>
            <Modal.Footer>
                <Button onClick={props.onClose}>Cancel</Button>
                <Button bsStyle="danger" onClick={props.onClear}>
                    Discard
                </Button>
            </Modal.Footer>
        </Modal>
    );
};

ClearChangesModal.propTypes = {
    onClear: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired
};

export default ClearChangesModal;
```

gives me,

```
  10:16  error  Handler function for onHide prop key must be named handleHide    react/jsx-handler-names
  19:25  error  Handler function for onClick prop key must be named handleClick  react/jsx-handler-names
  20:42  error  Handler function for onClick prop key must be named handleClick  react/jsx-handler-names
```

which is wrong, I suppose.

For now I disabled `react/jsx-handler-names` rule in the component, but hope you can fix it soon. Thanks!

**Comment by Pathsofdesign** [comment]
Good call, I just ran across this myself. :1 thank you!
